src/facet_display/Facet.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react"
+import { contains } from "ramda"
+
+import { SearchFacetItem } from "./SearchFacetItem"
+import { BucketWithLabel } from "./FacetDisplay"
+
+const MAX_DISPLAY_COUNT = 5
+const FACET_COLLAPSE_THRESHOLD = 15
+
+interface Props {
+  name: string
+  title: string
+  results: BucketWithLabel[] | null
+  selected: string[]
+  onUpdate: React.ChangeEventHandler<HTMLInputElement>
+  expandedOnLoad: boolean
+}
+
+function SearchFacet(props: Props) {
+  const { name, title, results, selected, onUpdate, expandedOnLoad } = props
+
+  const [showFacetList, setShowFacetList] = useState(expandedOnLoad)
+  const [showAllFacets, setShowAllFacets] = useState(false)
+
+  const titleLineIcon = showFacetList ? "arrow_drop_down" : "arrow_right"
+
+  return results && results.length === 0 ? null : (
+    <div className="facets">
+      <button
+        className="filter-section-button"
+        type="button"
+        aria-expanded={showFacetList ? "true" : "false"}
+        onClick={() => setShowFacetList(!showFacetList)}
+      >
+        {title}
+        <i className={`material-icons ${titleLineIcon}`} aria-hidden="true">
```

Comment:
Will projects using `@mitodl/course-search-utils` always be using Material icons? We should provide classnames for projects to hook into to provide their styles, though these should probable be generic, e.g. `expanded-open-icon`, `expanded-closed-icon`.

I appreciate we're porting code here - perhaps a question for our next revision after successfully moved this functionality across.

Reply:
I think for now it's ok to  assume that we'll use material icons. We can allow for other icons to be used if the need arises later